networkx/generators/tests/test_classic.py

```diff
@@ -258,6 +265,12 @@ def test_dorogovtsev_goltsev_mendes_graph(self):
             7,
             create_using=nx.MultiGraph,
         )
+        pytest.raises(
+            nx.NetworkXError,
+            nx.dorogovtsev_goltsev_mendes_graph,
+            7,
+            create_using=nx.MultiDiGraph,
+        )
```

Comment:
While we're here, can we update these exception tests to use the pytest context manager? It's easier to read and the context is probably better in other ways too. Can be done to the two new ones and update the 2 already there.

```suggestion
        with pytest.raises(nx.NetworkXError):
            nx.dorogovtsev_goltsev_mendes_graph(7, create_using=nx.MultiDiGraph)
```

Reply:
That does look cleaner. Will do.